c10/core/DispatchKey.cpp

```diff
@@ -255,6 +261,7 @@ c10::DispatchKey parseDispatchKey(const std::string& k) {
       {"Conjugate", c10::DispatchKey::Conjugate},
       {"Negative", c10::DispatchKey::Negative},
       {"ZeroTensor", c10::DispatchKey::ZeroTensor},
+      {"CsrTranspose", c10::DispatchKey::CsrTranspose},
```

Comment:
Wouldn't it be more consistent to call this SparseCsc instead of CsrTranspose?

Reply:
I think not. At first, I was thinking about introducing SparseCsc dispatch key but since we are reusing SparseCsrCPU and SparseCsrCUDA dispatch keys/implementations for CSC layout, it would have a weird DispatchKeySets content:
`[SparseCsrCPU, SparseCsc]`. After https://github.com/pytorch/pytorch/pull/72827 , the content would be even more confusing: `[CPU, SparseCsr, SparseCsc]` that represents a CSC tensor case.

Currently, we have `[SparseCsrCPU, CsrTranspose]` and after #72827 it would be `[CPU, SparseCsr, CsrTranspose]` where `CsrTranspose` indicates that the tensor represents transposed CSR tensor that is equivalent to CSC.

I also considered renaming `CsrTranspose` to `Transpose` so that perhaps other layouts could use it but that would be too restricted for tensors with dimensionality greater than 2 as transpose can have multiple variants (dimensions swaps).

On the other hand, it looks like after #72827, that will drop CPU/CUDA suffices from the sparse CSR dispatch keys, it might be more consistent to introduce SparseCsc dispatch key in parallel to SparseCsr and CsrTranspose key would not be needed anymore (we'll save one dk).